sklearn/cluster/_optics.py

```diff
@@ -243,7 +245,13 @@ def fit(self, X, y=None):
         self : instance of OPTICS
             The instance.
         """
-        X = self._validate_data(X, dtype=float)
+
+        dtype = bool if self.metric in PAIRWISE_BOOLEAN_FUNCTIONS else float
+        if dtype == bool and X.dtype != bool:
+            msg = "Data was converted to boolean for metric %s" % self.metric
```

Comment:
```suggestion
            msg = "Data will be converted to boolean for metric %s" % self.metric
```

Reply:
Also we may add "to avoid this warning, you may convert the data prior to calling fit."

Reply:
Also, please use f-strings instead of the `%` syntax

Reply:
thanks for the feedback @NicolasHug, dont know how I missed adding the regression tests, but they are there now.